tests/e2e/vmlistener_test.go

```diff
@@ -80,41 +74,77 @@ func (s *VMListenerTestParams) TearDownTest(c *C) {
 		out, err := dockercli.RemoveContainer(s.vm1, s.containerName)
 		c.Assert(err, IsNil, Commentf(out))
 	}
-	out, err := dockercli.DeleteVolume(s.vm1, s.volumeName)
-	c.Assert(err, IsNil, Commentf(out))
 }
 
 var _ = Suite(&VMListenerTestParams{})
 
 // Test vmdkops service restart
-// 1. Setup: Create a volume
-// 2. Setup: Attach it to a container
-// 3. Restart the vmdkops service
-// 4. Verification: volume stays as attached
+// 1. Create a volume
+// 2. Attach the volume to a container
+// 3. Verification: volume is in attached status
+// 4. Restart the vmdkops service
+// 5. Verification: volume stays as attached
+// 6. Remove the container
+// 7. Remove the volume
 func (s *VMListenerTestParams) TestServiceRestart(c *C) {
 	misc.LogTestStart(c.TestName())
 
+	// Create the volume
+	out, err := dockercli.CreateVolume(s.vm1, s.volumeName)
```

Comment:
@shaominchen now this test adds some inconsistency what we have decided earlier .. as each test is creating volume why not use setupTest fixture. 

For adding new test requires these amount of refactor, could be avoided by creating a separate file for vsan failure test